powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/util/PowerAuthVersionUtil.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * PowerAuth integration libraries for RESTful API applications, examples and
+ * related software components
+ *
+ * Copyright (C) 2023 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.rest.api.spring.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Utility class to assist with PowerAuth version checks and related functionalities.
+ *
+ * @author Jan Dusil, jan.dusil@wultra.com
+ */
+@Slf4j
+public class PowerAuthVersionUtil {
+
+    /**
+     * Enumeration representing supported PowerAuth versions.
+     */
+    private enum PowerAuthVersion {
```

Comment:
Why have you decided to use enum instead of String constants? What is the main benefit?

Reply:
IMHO encapsulation, code readability, centralized logic

Reply:
Well, I do not agree. You are writing about encapsulation, but comparing strings anyway https://github.com/wultra/powerauth-restful-integration/pull/443/files#diff-a1bf9cde4d0504f17919827964f48a9258b249173870897716f19af096d5e6e9R160
```java
!version.equals(PowerAuthVersion.V3_0.getVersionString())
```
It should be at least:
```java
PowerAuthVersion.fromString(version) != PowerAuthVersion.V3_0
```
But it still looks too complicated to me.
If you want to make it elegant, personally I would extract this:
```java
//!version.equals(PowerAuthVersion.V3_0.getVersionString())
...
!isVersion3_0(version);
//...

private static boolean isVersion3_0(final String version) {
    return V3_0.equals(version);
}
```
I still do not see any benefit from adding another layer of complexity, for example here https://github.com/wultra/powerauth-restful-integration/pull/443/files#diff-a1bf9cde4d0504f17919827964f48a9258b249173870897716f19af096d5e6e9R137
```java
return PowerAuthVersion.fromString(version) == null;
```
How is it better than?
```java
private static final Set<String> SUPPORTED_VERSIONS = Set.of("3.2", "3.1", "3.0");

private static boolean isUnsupportedVersion(final String version) {
    return SUPPORTED_VERSIONS.contains(version);
}
```
In this way, we avoid handling `null` anyway.